src/participant/prompts/intent.ts

```diff
@@ -0,0 +1,78 @@
+import * as vscode from 'vscode';
+
+import { getHistoryMessages } from './history';
+
+export type PromptIntent = 'Query' | 'Schema' | 'Docs' | 'Default';
+
+export class IntentPrompt {
+  static getAssistantPrompt(): vscode.LanguageModelChatMessage {
+    const prompt = `You are a MongoDB expert.
+Your task is to help guide a conversation with a user to the correct handler.
+You will be provided a conversation and your task is to determine the intent of the user.
+The intent handlers are:
+- Query
+- Schema
+- Docs
+- Default
+Rules:
+1. Respond only with the intent handler.
+2. Use the "Query" intent handler when the user is asking for code that relates to a specific collection.
+3. Use the "Docs" intent handler when the user is asking a question that involves MongoDB documentation.
```

Comment:
Did you try this out? I am wondering how the model interprets this instruction. Should users explicitly say find in the documentation? Or they can just ask "How to..." questions?

Reply:
The model did successfully route a fairly docs related question to the docs handler in my manual testing. Haven't been that extensive to give a good answer on when it does and when it doesn't. We'd need more accuracy tests.